single_tech_samples/synapseanalytics/sample3_integration_testing/README.md

```diff
@@ -0,0 +1,180 @@
+- [jq](https://stedolan.github.io/jq/)
+- [makepasswd](https://manpages.debian.org/stretch/makepasswd/makepasswd.1.en.html)
+
+#### Software pre-requisites if you use dev container
+
+- [Docker](https://www.docker.com/)
+- [VSCode](https://code.visualstudio.com/)
+- [Visual Studio Code Remote Development Extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.vscode-remote-extensionpack)
+
+  It is strongly recommended to use dev container for the deployment to avoid environment related issues.
+
+### Setup and Deployment
+
+> **IMPORTANT NOTE:** As with all Azure Deployments, this will **incur associated costs**. Remember to teardown all related resources after use to avoid unnecessary costs. See [here](#deployed-resources) for list of deployed resources. See [here](#clean-up) for information on the clean_up script.
+> This deployment was tested using WSL 2 (Ubuntu 20.04) and Debian GNU/Linux 9.9 (stretch)
+
+1. **Initial Setup**
+   - Ensure that:
+      - You have the desired Azure Cloud set. To change from the default (`AzureCloud`), run
+
+        ```bash
+        az cloud set -n <AZURE_CLOUD_NAME>
+        ```
+
+          - For example, if you want to use the Azure US Gov cloud, run
+
+            ```bash
+            az cloud set -n AzureUSGovernment
+            ```
+
+          - To see all of the Azure Cloud options, run
+
+            ```bash
+            az cloud list --output table
+            ```
+
+      - You are logged in to the Azure CLI. To login, run
+
+        ```bash
+        az login
+        ```
+
+      - Azure CLI is targeting the Azure Subscription you want to deploy the resources to. To set target Azure Subscription, run
+
+        ```bash
+        az account set -s <AZURE_SUBSCRIPTION_ID>
+        ```
+
+       Set the following environment variables:
+      - **AZURE_LOCATION** - Azure location to deploy resources. *Default*: `westus`.
+      - **AZURE_SUBSCRIPTION_ID** - Azure subscription id to use to deploy resources. *Default*: default azure subscription. To see your default, run `az account list`.
+      - **DEPLOYMENT_ID** - string appended to all resource names. This is to ensure uniqueness of azure resource names. *Default*: random five character string.
+      - **SYNAPSE_SQL_PASSWORD** - Password of the Synapse SQL instance. *Default*: random string.
+
+   - If you are using **dev container**, follow the below steps:
+      - Rename `.envtemplate` under ".devcontainer" folder to `devcontainer.env` and update the values as mentioned above instead of setting those as environment variables.
+      - Open the project inside the vscode dev container (see details [here](docs/devcontainer.md)).
+
+     To further customize the solution, set parameters in `arm.parameters` files located in the `infrastructure` folder.
+
+1. **Deploy Azure resources**
+   - `cd` into the `single_tech_samples/synapseanalytics/sample3_integration_testing` folder of the repo
+
+   - Run `./deploy.sh`.
```

Comment:
in the devcontainer, i got the below error

![image](https://user-images.githubusercontent.com/9535962/207906599-43026325-0087-4c00-ab86-04a1ed309b74.png)

I had to run the command `chmod u+x deploy.sh` to ensure that my user was authorized to execute it

Reply:
I had to do the same for `deploy_infrastructure.sh`, this may be worth taking care within the devcontainer

Reply:
Ultimately I ran this command to make sure i could execute all sh files `chmod u+x scripts/*.sh`

Reply:
so after doing some reading on devcontainers, I think the best path forward is to ensure use the "postCreateCommand" in the devcontainer.json.   

```"postCreateCommand": "chmod -R 774 scripts"```

This does result in the user having to enter any character to be able to access the devcontainer terminal.  technically, if the scripts are committed using the appropriate permissions, I don't think this command would be needed. But given that I ran into the error of permissions, it tells me that IF the scripts are written on Windows and committed to the repo, AND if the consumers of the repo clone it using UNIX, then those users would likely encounter this problem without having this command here. 

Reply:
Thank you for surfacing this and finding a solution!! I've updated the dev container to include this.

Reply:
I added the command in the devcontainer.json file and now see an error when rebuilding the container.

![image](https://user-images.githubusercontent.com/5297748/208698701-777ae522-bcd9-45d0-b8b7-0a45ca893055.png)

<img width="656" alt="image" src="https://user-images.githubusercontent.com/5297748/208698660-103491ec-22ad-45e3-93ce-6e8f6936c3f5.png">